Sources/SwiftKafka/RDKafka/RDKafkaClient.swift

```diff
@@ -97,33 +97,45 @@ final class RDKafkaClient: Sendable {
     /// - Parameter newMessageID: ID that was assigned to the `message`.
     /// - Parameter topicConfig: The ``KafkaTopicConfiguration`` used for newly created topics.
     /// - Parameter topicHandles: Topic handles that this client uses to produce new messages
-    func produce(
-        message: KafkaProducerMessage,
+    func produce<Key, Value>(
+        message: KafkaProducerMessage<Key, Value>,
         newMessageID: UInt,
         topicConfig: KafkaTopicConfiguration,
         topicHandles: RDKafkaTopicHandles
     ) throws {
-        let keyBytes: [UInt8]?
-        if var key = message.key {
-            keyBytes = key.readBytes(length: key.readableBytes)
-        } else {
-            keyBytes = nil
-        }
-
-        let responseCode = try message.value.withUnsafeReadableBytes { valueBuffer in
-            return try topicHandles.withTopicHandlePointer(topic: message.topic, topicConfig: topicConfig) { topicHandle in
-                // Pass message over to librdkafka where it will be queued and sent to the Kafka Cluster.
-                // Returns 0 on success, error code otherwise.
-                return rd_kafka_produce(
-                    topicHandle,
-                    message.partition.rawValue,
-                    RD_KAFKA_MSG_F_COPY,
-                    UnsafeMutableRawPointer(mutating: valueBuffer.baseAddress),
-                    valueBuffer.count,
-                    keyBytes,
-                    keyBytes?.count ?? 0,
-                    UnsafeMutableRawPointer(bitPattern: newMessageID)
-                )
+        let responseCode = try message.value.withUnsafeRawBufferPointer { valueBuffer in
```

Comment:
Can we create method in message?
```
func withUnsafeKeyValueRawBufferPointer(_ closure: (Buffer?, Buffer)) {
    let responseCode = try value.withUnsafeRawBufferPointer { valueBuffer in
        if let key {
            key.withUnsafeRawBufferPointer { keyBuffer in
                closure(keyBuffer, valueBuffer)
            }
        } else {
            closure(nil, valueBuffer)
        }
    }
}
```
Then we could avoid call `key.withUnsafeRawBufferPointer` for every topic. Especially, it might make sense for custom types.

Reply:
Ou, sorry, I see that it is only for one topic.